core/src/main/java/cz/xtf/core/openshift/OpenShift.java

```diff
@@ -165,7 +179,25 @@ public static OpenShift get(String masterUrl, String namespace, String token) {
 
         setupTimeouts(openShiftConfig);
 
-        return new OpenShift(openShiftConfig);
+        return get(openShiftConfig);
+    }
+
+    private static OpenShift get(OpenShiftConfig openShiftConfig) {
+        OpenShift openshift;
+
+        // check whether such a client already exists
+        Optional<OpenShift> optionalOpenShift = namespaceToOpenshiftClientMap
+                .get(openShiftConfig.getNamespace()).stream()
+                .filter(oc -> isEqualOpenshiftConfig(openShiftConfig, oc.getConfiguration()))
+                .findFirst();
```

Comment:
Can't this introduce some intermittent odd behaviour?

So I have test where I have under one namespace different OpenShift clients - typically `master()` and `admin`. Doesn't `findFirst()` return those clients randomly? 

This brings me back to my objection that `namespaceToOpenshiftClientMap` must use as a key OpenShiftConfig. Then username will be taken into account correctly.